Linux/O_STM/DOXY/man/man3/SLIGO_W.3
.TH "SLIGO_W" 3 "Sun Apr 1 2018" "C++ Software transactional Memory" \" -*- nroff -*-
.ad l
.nh
.SH NAME
SLIGO_W \- 
.SH SYNOPSIS
.br
.PP
.PP
\fC#include <SLIGO_W\&.h>\fP
.PP
Inherits \fBWAREHOUSE\fP\&.
.SS "Public Member Functions"

.in +1c
.ti -1c
.RI "virtual void \fBcopy\fP (std::shared_ptr< \fBOSTM\fP > to, std::shared_ptr< \fBOSTM\fP > from)"
.br
.RI "\fIcopy function, make deep copy of the object/pointer \fP"
.ti -1c
.RI "virtual std::shared_ptr< \fBOSTM\fP > \fBgetBaseCopy\fP (std::shared_ptr< \fBOSTM\fP > object)"
.br
.RI "\fIgetBaseCopy function, make deep copy of the object/pointer and Return a new BANK* type object \fP"
.ti -1c
.RI "virtual int \fBGetNumber_of_alcatel\fP ()"
.br
.ti -1c
.RI "virtual int \fBGetNumber_of_huawei\fP ()"
.br
.ti -1c
.RI "virtual int \fBGetNumber_of_iphones\fP ()"
.br
.ti -1c
.RI "virtual int \fBGetNumber_of_nokia\fP ()"
.br
.ti -1c
.RI "virtual int \fBGetNumber_of_samsung\fP ()"
.br
.ti -1c
.RI "virtual int \fBGetNumber_of_sony\fP ()"
.br
.ti -1c
.RI "virtual std::string \fBGetShop_address\fP ()"
.br
.ti -1c
.RI "virtual std::string \fBGetShop_name\fP ()"
.br
.ti -1c
.RI "\fBSLIGO_W\fP \fBoperator=\fP (const \fBSLIGO_W\fP &orig)"
.br
.ti -1c
.RI "virtual void \fBSetNumber_of_alcatel\fP (int \fB_number_of_alcatel\fP)"
.br
.ti -1c
.RI "virtual void \fBSetNumber_of_huawei\fP (int \fB_number_of_huawei\fP)"
.br
.ti -1c
.RI "virtual void \fBSetNumber_of_iphones\fP (int \fB_number_of_iphones\fP)"
.br
.ti -1c
.RI "virtual void \fBSetNumber_of_nokia\fP (int \fB_number_of_nokia\fP)"
.br
.ti -1c
.RI "virtual void \fBSetNumber_of_samsung\fP (int \fB_number_of_samsung\fP)"
.br
.ti -1c
.RI "virtual void \fBSetNumber_of_sony\fP (int \fB_number_of_sony\fP)"
.br
.ti -1c
.RI "virtual void \fBSetShop_address\fP (std::string \fB_shop_address\fP)"
.br
.ti -1c
.RI "virtual void \fBSetShop_name\fP (std::string \fB_shop_name\fP)"
.br
.ti -1c
.RI "\fBSLIGO_W\fP ()"
.br
.ti -1c
.RI "\fBSLIGO_W\fP (std::string address, std::string shop_name, int iphone, int samsung, int sony, int huawei, int nokia, int alcatel)"
.br
.ti -1c
.RI "\fBSLIGO_W\fP (std::shared_ptr< \fBWAREHOUSE\fP > obj, int _version, int _unique_id)"
.br
.ti -1c
.RI "\fBSLIGO_W\fP (const \fBSLIGO_W\fP &orig)"
.br
.ti -1c
.RI "virtual void \fBtoString\fP ()"
.br
.RI "\fI_cast, is use to cast bak the std::shared_ptr<OSTM> to the required type \fP"
.ti -1c
.RI "virtual \fB~SLIGO_W\fP ()"
.br
.in -1c
.SS "Private Attributes"

.in +1c
.ti -1c
.RI "int \fB_number_of_alcatel\fP"
.br
.ti -1c
.RI "int \fB_number_of_huawei\fP"
.br
.ti -1c
.RI "int \fB_number_of_iphones\fP"
.br
.ti -1c
.RI "int \fB_number_of_nokia\fP"
.br
.ti -1c
.RI "int \fB_number_of_samsung\fP"
.br
.ti -1c
.RI "int \fB_number_of_sony\fP"
.br
.ti -1c
.RI "std::string \fB_shop_address\fP"
.br
.ti -1c
.RI "std::string \fB_shop_name\fP"
.br
.in -1c
.SH "Detailed Description"
.PP 
Inherit from \fBWAREHOUSE\fP 
.PP
Definition at line 19 of file SLIGO_W\&.h\&.
.SH "Constructor & Destructor Documentation"
.PP 
.SS "SLIGO_W::SLIGO_W ()\fC [inline]\fP"
Constructor 
.PP
Definition at line 24 of file SLIGO_W\&.h\&.
.PP
References _number_of_alcatel, _number_of_huawei, _number_of_iphones, _number_of_nokia, _number_of_samsung, _number_of_sony, _shop_address, and _shop_name\&.
.PP
Referenced by getBaseCopy(), and SLIGO_W()\&.
.PP
.nf
24               : WAREHOUSE(){
25         
26         this->_shop_address = "Sligo River Street";
27         this->_shop_name = "SLIGO S_WAREHOUSE";
28         this->_number_of_iphones = 200;
29         this->_number_of_samsung = 200;
30         this->_number_of_sony = 200;
31         this->_number_of_huawei = 200;
32         this->_number_of_nokia = 200;
33         this->_number_of_alcatel = 200;
34     };
.fi
.SS "SLIGO_W::SLIGO_W (std::string address, std::string shop_name, int iphone, int samsung, int sony, int huawei, int nokia, int alcatel)\fC [inline]\fP"
Custom constructor 
.PP
Definition at line 38 of file SLIGO_W\&.h\&.
.PP
References _number_of_alcatel, _number_of_huawei, _number_of_iphones, _number_of_nokia, _number_of_samsung, _number_of_sony, _shop_address, and _shop_name\&.
.PP
.nf
38                                                                                                                           : WAREHOUSE(){
39         /*
40          * copy over values
41          */
42         this->_shop_address = address;
43         this->_shop_name = shop_name;
44         this->_number_of_iphones = iphone;
45         this->_number_of_samsung = samsung;
46         this->_number_of_sony = sony;
47         this->_number_of_huawei = huawei;
48         this->_number_of_nokia = nokia;
49         this->_number_of_alcatel = alcatel;
50         
51     };
.fi
.SS "SLIGO_W::SLIGO_W (std::shared_ptr< \fBWAREHOUSE\fP > obj, int _version, int _unique_id)\fC [inline]\fP"
Custom constructor, used by the library for deep copying 
.PP
Definition at line 55 of file SLIGO_W\&.h\&.
.PP
References _number_of_alcatel, _number_of_huawei, _number_of_iphones, _number_of_nokia, _number_of_samsung, _number_of_sony, _shop_address, _shop_name, and SLIGO_W()\&.
.PP
.nf
55                                                                        : WAREHOUSE(_version, _unique_id){
56         /*
57          * copy over values
58          */
59         this->_shop_address = obj->GetShop_address();
60         this->_shop_name = obj->GetShop_name();
61         this->_number_of_iphones = obj->GetNumber_of_iphones();
62         this->_number_of_samsung = obj->GetNumber_of_samsung();
63         this->_number_of_sony = obj->GetNumber_of_sony();
64         this->_number_of_huawei = obj->GetNumber_of_huawei();
65         this->_number_of_nokia = obj->GetNumber_of_nokia();
66         this->_number_of_alcatel = obj->GetNumber_of_alcatel();
67     }
.fi
.SS "SLIGO_W::SLIGO_W (const \fBSLIGO_W\fP & orig)"
Copy constructor 
.PP
Definition at line 15 of file SLIGO_W\&.cpp\&.
.PP
.nf
15                                     {
16 }
.fi
.SS "SLIGO_W::~SLIGO_W ()\fC [virtual]\fP"
de-constructor 
.PP
Definition at line 12 of file SLIGO_W\&.cpp\&.
.PP
Referenced by operator=()\&.
.PP
.nf
12                   {
13 }
.fi
.SH "Member Function Documentation"
.PP 
.SS "void SLIGO_W::copy (std::shared_ptr< \fBOSTM\fP > to, std::shared_ptr< \fBOSTM\fP > from)\fC [virtual]\fP"

.PP
copy function, make deep copy of the object/pointer 
.PP
\fBParameters:\fP
.RS 4
\fIobjTO\fP is a BANK* type object casted back from std::shared_ptr<OSTM> 
.br
\fIobjFROM\fP is a BANK* type object casted back from std::shared_ptr<OSTM> 
.RE
.PP

.PP
Reimplemented from \fBOSTM\fP\&.
.PP
Definition at line 35 of file SLIGO_W\&.cpp\&.
.PP
References _shop_address\&.
.PP
Referenced by operator=()\&.
.PP
.nf
35                                                                   {
36 
37     std::shared_ptr<SLIGO_W> objTO = std::dynamic_pointer_cast<SLIGO_W>(to);
38     std::shared_ptr<SLIGO_W> objFROM = std::dynamic_pointer_cast<SLIGO_W>(from);
39     objTO->_shop_address = objFROM->GetShop_address();
40     objTO->_shop_name = objFROM->GetShop_name();
41     objTO->_number_of_iphones = objFROM->GetNumber_of_iphones();
42     objTO->_number_of_samsung = objFROM->GetNumber_of_samsung();
43     objTO->_number_of_sony = objFROM->GetNumber_of_sony();
44     objTO->_number_of_huawei = objFROM->GetNumber_of_huawei();
45     objTO->_number_of_nokia = objFROM->GetNumber_of_nokia();
46     objTO->_number_of_alcatel = objFROM->GetNumber_of_alcatel();
47     objTO->Set_Unique_ID(objFROM->Get_Unique_ID());
48     objTO->Set_Version(objFROM->Get_Version());
49     
50    
51 }
.fi
.SS "std::shared_ptr< \fBOSTM\fP > SLIGO_W::getBaseCopy (std::shared_ptr< \fBOSTM\fP > object)\fC [virtual]\fP"

.PP
getBaseCopy function, make deep copy of the object/pointer and Return a new BANK* type object 
.PP
\fBParameters:\fP
.RS 4
\fIobjTO\fP is a \fBBANK\fP type pointer for casting 
.br
\fIobj\fP is a BANK* return type 
.RE
.PP

.PP
Reimplemented from \fBOSTM\fP\&.
.PP
Definition at line 22 of file SLIGO_W\&.cpp\&.
.PP
References SLIGO_W()\&.
.PP
Referenced by operator=()\&.
.PP
.nf
23 {
24 
25     std::shared_ptr<WAREHOUSE> objTO = std::dynamic_pointer_cast<WAREHOUSE>(object);
26     std::shared_ptr<WAREHOUSE> obj(new SLIGO_W(objTO, object->Get_Version(),object->Get_Unique_ID()));
27     std::shared_ptr<OSTM> ostm_obj = std::dynamic_pointer_cast<OSTM>(obj);                                  
28     return ostm_obj;
29 }
.fi
.SS "int SLIGO_W::GetNumber_of_alcatel ()\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 73 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_alcatel\&.
.PP
Referenced by operator=(), and toString()\&.
.PP
.nf
73                                  {
74     return _number_of_alcatel;
75 }
.fi
.SS "int SLIGO_W::GetNumber_of_huawei ()\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 89 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_huawei\&.
.PP
Referenced by operator=(), and toString()\&.
.PP
.nf
89                                 {
90     return _number_of_huawei;
91 }
.fi
.SS "int SLIGO_W::GetNumber_of_iphones ()\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 113 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_iphones\&.
.PP
Referenced by operator=(), and toString()\&.
.PP
.nf
113                                  {
114     return _number_of_iphones;
115 }
.fi
.SS "int SLIGO_W::GetNumber_of_nokia ()\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 81 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_nokia\&.
.PP
Referenced by operator=(), and toString()\&.
.PP
.nf
81                                {
82     return _number_of_nokia;
83 }
.fi
.SS "int SLIGO_W::GetNumber_of_samsung ()\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 105 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_samsung\&.
.PP
Referenced by operator=(), and toString()\&.
.PP
.nf
105                                  {
106     return _number_of_samsung;
107 }
.fi
.SS "int SLIGO_W::GetNumber_of_sony ()\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 97 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_sony\&.
.PP
Referenced by operator=(), and toString()\&.
.PP
.nf
97                               {
98     return _number_of_sony;
99 }
.fi
.SS "std::string SLIGO_W::GetShop_address ()\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 129 of file SLIGO_W\&.cpp\&.
.PP
References _shop_address\&.
.PP
Referenced by operator=(), and toString()\&.
.PP
.nf
129                                   {
130     return _shop_address;
131 }
.fi
.SS "std::string SLIGO_W::GetShop_name ()\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 121 of file SLIGO_W\&.cpp\&.
.PP
References _shop_name\&.
.PP
Referenced by operator=(), and toString()\&.
.PP
.nf
121                                {
122     return _shop_name;
123 }
.fi
.SS "\fBSLIGO_W\fP SLIGO_W::operator= (const \fBSLIGO_W\fP & orig)\fC [inline]\fP"
Operator 
.PP
Definition at line 75 of file SLIGO_W\&.h\&.
.PP
References _number_of_alcatel, _number_of_huawei, _number_of_iphones, _number_of_nokia, _number_of_samsung, _number_of_sony, _shop_address, _shop_name, copy(), getBaseCopy(), GetNumber_of_alcatel(), GetNumber_of_huawei(), GetNumber_of_iphones(), GetNumber_of_nokia(), GetNumber_of_samsung(), GetNumber_of_sony(), GetShop_address(), GetShop_name(), SetNumber_of_alcatel(), SetNumber_of_huawei(), SetNumber_of_iphones(), SetNumber_of_nokia(), SetNumber_of_samsung(), SetNumber_of_sony(), SetShop_address(), SetShop_name(), toString(), and ~SLIGO_W()\&.
.PP
.nf
75 {};
.fi
.SS "void SLIGO_W::SetNumber_of_alcatel (int _number_of_alcatel)\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 69 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_alcatel\&.
.PP
Referenced by operator=()\&.
.PP
.nf
69                                                          {
70     this->_number_of_alcatel = _number_of_alcatel;
71 }
.fi
.SS "void SLIGO_W::SetNumber_of_huawei (int _number_of_huawei)\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 85 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_huawei\&.
.PP
Referenced by operator=()\&.
.PP
.nf
85                                                        {
86     this->_number_of_huawei = _number_of_huawei;
87 }
.fi
.SS "void SLIGO_W::SetNumber_of_iphones (int _number_of_iphones)\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 109 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_iphones\&.
.PP
Referenced by operator=()\&.
.PP
.nf
109                                                          {
110     this->_number_of_iphones = _number_of_iphones;
111 }
.fi
.SS "void SLIGO_W::SetNumber_of_nokia (int _number_of_nokia)\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 77 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_nokia\&.
.PP
Referenced by operator=()\&.
.PP
.nf
77                                                      {
78     this->_number_of_nokia = _number_of_nokia;
79 }
.fi
.SS "void SLIGO_W::SetNumber_of_samsung (int _number_of_samsung)\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 101 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_samsung\&.
.PP
Referenced by operator=()\&.
.PP
.nf
101                                                          {
102     this->_number_of_samsung = _number_of_samsung;
103 }
.fi
.SS "void SLIGO_W::SetNumber_of_sony (int _number_of_sony)\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 93 of file SLIGO_W\&.cpp\&.
.PP
References _number_of_sony\&.
.PP
Referenced by operator=()\&.
.PP
.nf
93                                                    {
94     this->_number_of_sony = _number_of_sony;
95 }
.fi
.SS "void SLIGO_W::SetShop_address (std::string _shop_address)\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 125 of file SLIGO_W\&.cpp\&.
.PP
References _shop_address\&.
.PP
Referenced by operator=()\&.
.PP
.nf
125                                                      {
126     this->_shop_address = _shop_address;
127 }
.fi
.SS "void SLIGO_W::SetShop_name (std::string _shop_name)\fC [virtual]\fP"

.PP
Reimplemented from \fBWAREHOUSE\fP\&.
.PP
Definition at line 117 of file SLIGO_W\&.cpp\&.
.PP
References _shop_name\&.
.PP
Referenced by operator=()\&.
.PP
.nf
117                                                {
118     this->_shop_name = _shop_name;
119 }
.fi
.SS "void SLIGO_W::toString ()\fC [virtual]\fP"

.PP
_cast, is use to cast bak the std::shared_ptr<OSTM> to the required type toString function, displays the object values in formatted way 
.PP
Reimplemented from \fBOSTM\fP\&.
.PP
Definition at line 62 of file SLIGO_W\&.cpp\&.
.PP
References OSTM::Get_Unique_ID(), OSTM::Get_Version(), GetNumber_of_alcatel(), GetNumber_of_huawei(), GetNumber_of_iphones(), GetNumber_of_nokia(), GetNumber_of_samsung(), GetNumber_of_sony(), GetShop_address(), and GetShop_name()\&.
.PP
Referenced by operator=()\&.
.PP
.nf
63 {
64     std::cout << "\n" <<  this->GetShop_name() << "\nUnique ID : " << this->Get_Unique_ID() << "\nShop Name : "  << this->GetShop_name() << "\nShop Address : " << this->GetShop_address() << "\nNo\&. Iphones : " << this->GetNumber_of_iphones() << "\nNo\&. Samsung : " << this->GetNumber_of_samsung() << "\nNo\&. Sony : " << this->GetNumber_of_sony() << "\nNo\&. Huawei : " << this->GetNumber_of_huawei() << "\nNo\&. Nokia : " << this->GetNumber_of_nokia() << "\nNo\&. Alcatel : " << this->GetNumber_of_alcatel() << "\nVersion number : " << this->Get_Version() << std::endl;
65 }
.fi
.SH "Member Data Documentation"
.PP 
.SS "int SLIGO_W::_number_of_alcatel\fC [private]\fP"

.PP
Definition at line 117 of file SLIGO_W\&.h\&.
.PP
Referenced by GetNumber_of_alcatel(), operator=(), SetNumber_of_alcatel(), and SLIGO_W()\&.
.SS "int SLIGO_W::_number_of_huawei\fC [private]\fP"

.PP
Definition at line 115 of file SLIGO_W\&.h\&.
.PP
Referenced by GetNumber_of_huawei(), operator=(), SetNumber_of_huawei(), and SLIGO_W()\&.
.SS "int SLIGO_W::_number_of_iphones\fC [private]\fP"

.PP
Definition at line 112 of file SLIGO_W\&.h\&.
.PP
Referenced by GetNumber_of_iphones(), operator=(), SetNumber_of_iphones(), and SLIGO_W()\&.
.SS "int SLIGO_W::_number_of_nokia\fC [private]\fP"

.PP
Definition at line 116 of file SLIGO_W\&.h\&.
.PP
Referenced by GetNumber_of_nokia(), operator=(), SetNumber_of_nokia(), and SLIGO_W()\&.
.SS "int SLIGO_W::_number_of_samsung\fC [private]\fP"

.PP
Definition at line 113 of file SLIGO_W\&.h\&.
.PP
Referenced by GetNumber_of_samsung(), operator=(), SetNumber_of_samsung(), and SLIGO_W()\&.
.SS "int SLIGO_W::_number_of_sony\fC [private]\fP"

.PP
Definition at line 114 of file SLIGO_W\&.h\&.
.PP
Referenced by GetNumber_of_sony(), operator=(), SetNumber_of_sony(), and SLIGO_W()\&.
.SS "std::string SLIGO_W::_shop_address\fC [private]\fP"

.PP
Definition at line 110 of file SLIGO_W\&.h\&.
.PP
Referenced by copy(), GetShop_address(), operator=(), SetShop_address(), and SLIGO_W()\&.
.SS "std::string SLIGO_W::_shop_name\fC [private]\fP"

.PP
Definition at line 111 of file SLIGO_W\&.h\&.
.PP
Referenced by GetShop_name(), operator=(), SetShop_name(), and SLIGO_W()\&.

.SH "Author"
.PP 
Generated automatically by Doxygen for C++ Software transactional Memory from the source code\&.


Client application Tests for Linux/Test04/SLIGO_W.cpp

/* 
 * File:   SLIGO_W.cpp
 * Author: Zoltan Fuzesi
 * IT Carlow : C00197361
 *
 * Created on January 17, 2018, 8:02 PM
 */

#include "SLIGO_W.h"

SLIGO_W::~SLIGO_W() {
}

SLIGO_W::SLIGO_W(const SLIGO_W& orig) {
}
/*!
 * \brief getBaseCopy function, make deep copy of the object/pointer and Return a new BANK* type object
 * \param objTO is a BANK type pointer for casting
 * \param obj is a BANK* return type
 */
std::shared_ptr<OSTM> SLIGO_W::getBaseCopy(std::shared_ptr<OSTM> object)
{

    std::shared_ptr<WAREHOUSE> objTO = std::dynamic_pointer_cast<WAREHOUSE>(object);
    std::shared_ptr<WAREHOUSE> obj(new SLIGO_W(objTO, object->Get_Version(),object->Get_Unique_ID()));
    std::shared_ptr<OSTM> ostm_obj = std::dynamic_pointer_cast<OSTM>(obj);                                  
    return ostm_obj;
}
/*!
 * \brief copy function, make deep copy of the object/pointer
 * \param objTO is a BANK* type object casted back from std::shared_ptr<OSTM>
 * \param objFROM is a BANK* type object casted back from std::shared_ptr<OSTM>
 */
void SLIGO_W::copy(std::shared_ptr<OSTM> to, std::shared_ptr<OSTM> from){

    std::shared_ptr<SLIGO_W> objTO = std::dynamic_pointer_cast<SLIGO_W>(to);
    std::shared_ptr<SLIGO_W> objFROM = std::dynamic_pointer_cast<SLIGO_W>(from);
    objTO->_shop_address = objFROM->GetShop_address();
    objTO->_shop_name = objFROM->GetShop_name();
    objTO->_number_of_iphones = objFROM->GetNumber_of_iphones();
    objTO->_number_of_samsung = objFROM->GetNumber_of_samsung();
    objTO->_number_of_sony = objFROM->GetNumber_of_sony();
    objTO->_number_of_huawei = objFROM->GetNumber_of_huawei();
    objTO->_number_of_nokia = objFROM->GetNumber_of_nokia();
    objTO->_number_of_alcatel = objFROM->GetNumber_of_alcatel();
    objTO->Set_Unique_ID(objFROM->Get_Unique_ID());
    objTO->Set_Version(objFROM->Get_Version());
    
   
}
/*!
 * \brief _cast, is use to cast bak the std::shared_ptr<OSTM> to the required type
 */
//std::shared_ptr<SLIGO_W> SLIGO_W::_cast(std::shared_ptr<OSTM> _object){
//    
//    return static_cast<std::shared_ptr<SLIGO_W>>(_object);
//}
/*!
 *  \brief toString function, displays the object values in formatted way
 */
void SLIGO_W::toString()
{
    std::cout << "\n" <<  this->GetShop_name() << "\nUnique ID : " << this->Get_Unique_ID() << "\nShop Name : "  << this->GetShop_name() << "\nShop Address : " << this->GetShop_address() << "\nNo. Iphones : " << this->GetNumber_of_iphones() << "\nNo. Samsung : " << this->GetNumber_of_samsung() << "\nNo. Sony : " << this->GetNumber_of_sony() << "\nNo. Huawei : " << this->GetNumber_of_huawei() << "\nNo. Nokia : " << this->GetNumber_of_nokia() << "\nNo. Alcatel : " << this->GetNumber_of_alcatel() << "\nVersion number : " << this->Get_Version() << std::endl;
}



void SLIGO_W::SetNumber_of_alcatel(int _number_of_alcatel) {
    this->_number_of_alcatel = _number_of_alcatel;
}

int SLIGO_W::GetNumber_of_alcatel(){
    return _number_of_alcatel;
}

void SLIGO_W::SetNumber_of_nokia(int _number_of_nokia) {
    this->_number_of_nokia = _number_of_nokia;
}

int SLIGO_W::GetNumber_of_nokia(){
    return _number_of_nokia;
}

void SLIGO_W::SetNumber_of_huawei(int _number_of_huawei) {
    this->_number_of_huawei = _number_of_huawei;
}

int SLIGO_W::GetNumber_of_huawei(){
    return _number_of_huawei;
}

void SLIGO_W::SetNumber_of_sony(int _number_of_sony) {
    this->_number_of_sony = _number_of_sony;
}

int SLIGO_W::GetNumber_of_sony(){
    return _number_of_sony;
}

void SLIGO_W::SetNumber_of_samsung(int _number_of_samsung) {
    this->_number_of_samsung = _number_of_samsung;
}

int SLIGO_W::GetNumber_of_samsung(){
    return _number_of_samsung;
}

void SLIGO_W::SetNumber_of_iphones(int _number_of_iphones) {
    this->_number_of_iphones = _number_of_iphones;
}

int SLIGO_W::GetNumber_of_iphones(){
    return _number_of_iphones;
}

void SLIGO_W::SetShop_name(std::string _shop_name) {
    this->_shop_name = _shop_name;
}

std::string SLIGO_W::GetShop_name(){
    return _shop_name;
}

void SLIGO_W::SetShop_address(std::string _shop_address) {
    this->_shop_address = _shop_address;
}

std::string SLIGO_W::GetShop_address(){
    return _shop_address;
}



